### PR TITLE
Fix/python3

### PR DESCRIPTION
**Pull request by aljcalandra**

### CODE REVIEW

File: test/API/ZoombaAPITestsData.py [hunk 1]
```diff
@@ -336,8 +343,12 @@
...
                    "\n------------------" \
                    "\nKey: AssignedDate" \
-                   "\nExpected: 2015-12-16T14:21:58Z" \
-                   "\nActual: 1234-34-45" \
+                   "\nActual Date Not Correct Format:" \
+                   "\nExpected Formats: %Y-%m-%dT%H:%M:%S" \
+                   "\n                  %Y-%m-%dT%H:%M:%SZ" \
+                   "\n                  %Y-%m-%dT%H:%M:%S.%f" \
+                   "\n                  %Y-%m-%dT%H:%M:%S.%fZ" \
+                   "\nDate: 1234-34-45" \
```
Comment (KeithSmoland) [inline test/API/ZoombaAPITestsData.py]: Why do we have a hard coded example of a bad date?  Shouldn't we return their actual formatted date?

Comment (aljcalandra) [reply test/API/ZoombaAPITestsData.py]: This is the response we're expecting from the test case.

File: src/Zoomba/APILibrary.py [hunk 2]
```diff
@@ -275,24 +275,11 @@ def date_string_comparator(self, expected_date, actual_date, key, unmatched_keys
...
             return
         else:
-            try:
-                expected_utc = datetime.datetime.strptime(expected_date, '%Y-%m-%dT%H:%M:%S')
-                actual_utc = datetime.datetime.strptime(actual_date, '%Y-%m-%dT%H:%M:%S')
+            expected_utc = _date_format(expected_date, key, unmatched_keys_list, "Expected")
+            actual_utc = _date_format(actual_date, key, unmatched_keys_list, "Actual")
+            if expected_utc and actual_utc:
                 self.date_comparator(expected_utc, actual_utc, key, unmatched_keys_list)
-            except ValueError:
```
Comment (KeithSmoland) [inline src/Zoomba/APILibrary.py]: Why are we removing this whole block?  Was nothing valid?

File: src/Zoomba/APILibrary.py [hunk 3]
```diff
@@ -328,3 +315,37 @@ def generate_unmatched_keys_error_message(self, unmatched_keys):
                     keys_error_msg += str(key_error) + "\n"
             assert False, keys_error_msg + "\nPlease see differing value(s)"
 
+
+def _date_format(date_string, key, unmatched_keys_list, date_type, date_format=None):
```
Comment (KeithSmoland) [inline src/Zoomba/APILibrary.py]: Why do we need this whole block for python 3?

File: src/Zoomba/APILibrary.py (hunk 2 above)
Comment (aljcalandra) [reply src/Zoomba/APILibrary.py]: just abstraction to the _date_format() below

File: src/Zoomba/APILibrary.py (hunk 3 above)
Comment (m-hintz-42) [reply src/Zoomba/APILibrary.py]: To my knowledge this is not a python3 specific change - it's a bug fix to the date format error handling. 

I would prefer if these were different PRs though. They are entirely separate issues.

**Pull request closed without merging**